app/scripts/services/authorization.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("AuthorizationService", function($q, $cacheFactory, Logger, $interval, DataService){
+    function AuthorizationService() {
+      this.currentProject = null;
+      this.cachedRulesByProject = $cacheFactory('rulesCache', {
+        number: 10
+      });
+    }
+
+    // Transform data from:
+    // rules = {resources: ["jobs"], apiGroups: ["extensions"], verbs:["create","delete","get","list","update"]}
+    // into:
+    // normalizedRules = {"extensions/jobs": ["create","delete","get","list","update"]}
+    // Also while looping through the rules, check if user can create/update any resource on the 'Add to project' so the button will be displayed.
+    var normalizeRules = function(rules) {
+      var normalizedRules = {};
+      var canAddToProject = null;
+
+      // Check if resource name meets one of following conditions, since those resources can't be create/update via `Add to project` page:
+      //  - 'projectrequests'
+      //  - subresource taht contains '/', eg: 'builds/source', 'builds/logs', ...
+      //  - resource ending with 'review' string: 'localsubjectaccessreviews', 'selfsubjectrulesreviews', 'subjectaccessreviews' 
+      var checkResource = function(resource) {
```

Comment:
Could probably create `checkResource()` outside of `normalizeRules()`, since it does the same thing every time called (no gain in recreating it every time `normalizeRules()` is called)